### ISSUE TITLE
Is there a onnx object detection model available?

### ISSUE THREAD

**Issue by kumardesappan** [onnx/models#17]
Hi,
Is there a  pre-trained onnx object detection model available for reference (SSD, Faster-RCNN, RetinaNet or Mask RCNN)?

Thanks and Regards,
Kumar.D

**Comment by houseroad** [comment]
We are working on this. To import these models to ONNX model zoo, we may need to introduce some new ops into ONNX spec.

**Comment by kumardesappan** [comment]
@houseroad Thanks for the update. Do you have any tentative timeline for this?

**Comment by houseroad** [comment]
Our plan is to add these models in 2~3 weeks.

**Comment by prasanthpul** [comment]
@houseroad Can you share list of models you plan to add?

**Comment by houseroad** [comment]
Our plan is to convert the models from https://github.com/caffe2/models.

bvlc_googlenet, bvlc_reference_caffenet, bvlc_reference_rcnn_ilsvrc13, finetune_flickr_style will be converted.

And many models in sytle_transfer and detectron (Faster-RCNN included) folders are also in our plan.

**Comment by 0wu** [comment]
Is there a ticket where we can peek the new Ops for detection?

**Comment by houseroad** [comment]
Will send the PR including BBoxTransform, BoxWithNMSLimit, and GenerateProposals which are need by Faster-RCNN soon.

**Comment by teichgraf** [comment]
Any update on this like an ETA? :)

**Comment by 0wu** [comment]
@houseroad any updates? or possibility to contribute code for this?

**Comment by bestlin** [comment]
@houseroad do you have some plans for the detection functions for SSD?

**Comment by rafisef** [comment]
@houseroad I know this has been asked before, but any updates on these models? Any plans for Mask RCNN?

**Comment by houseroad** [comment]
Yes, we do, still working on it, just low bandwidth.

**Comment by brantPTS** [comment]
Thanks for asking about this and for the updates. Any new information on status would be great.

**Comment by qigtang** [comment]
@houseroad any status update on mask-rcnn export?

**Comment by houseroad** [comment]
I think @linkerzhang is working on it. Any update?

**Comment by kubansal** [comment]
Hi @linkerzhang did you get a chance to have a look on the same

**Comment by imranparuk** [comment]
Hi, any update on this?

**Comment by daquexian** [comment]
@imranparuk Faster R-CNN, Mask R-CNN, yolov3 and SSD has already been introduced in model zoo.

**Comment by imranparuk** [comment]
@daquexian what about Retinanet?

**Comment by imranparuk** [comment]
I am guessing you need the model to be traceable with pytorch. Currently the Retinanet model isn't traceable due to the BoxList (I'm not 100% sure of this, just looking at the issue board for the maskedrcnn bechmark project)? However if its been solved for the Faster-RCNN and Masked-RCNN, surely it shouldn't be an issue for Retinanet?

**Comment by SnowRipple** [comment]
@daquexian  I've been trying to convert RetinaNet to onnx for couple days now but without success. Is RetinaNet officially supported by ONNX now?

**Comment by imranparuk** [comment]
@SnowRipple Doesn't seem to be the case, from the looks of things the models need to be traceable by torch jit (this doesn't guarantee it can be converted into onnx I don't think). 

It has been accomplished by others for other projects as there are onnx models for papers like Faster-RCNN etc. These models make use of similar classes, however there doesn't seem to be any documentation on how to accomplish this. 

If you do however find anything, please let me know!

**Comment by SnowRipple** [comment]
@imranparuk hopefully it is just a matter of time, since most of the Detectron models are already supported, so the RetinaNet should be too. 

I agree that a lot of ops that were created for the sake of other detection models can be used in RetiaNet again, however there are still ops that seem to be specific to RetinaNet only.

Personally I find it surprising that such a popular detection model is not supported yet, whereas other (much older with worse performance) models are.

**Comment by imranparuk** [comment]
@SnowRipple Yeah, I also find it surprising that they have left it out. 

Possibly our little chat here will send so many notifications the maintainers, they will rush to fix this :laughing: (at-least inform us on how we can go about doing it) 

Otherwise converting myself seems out of the question, my level of understanding of the original facebook project is very limited at this point.

**Comment by azuryl** [comment]
@houseroad  do you have Retinanet   onnx
